src/components/Common/ProposalPreviewCard/ProposalPreviewCard.tsx

```diff
@@ -24,37 +24,66 @@ import VoteModule from './VoteModule'
 enum Variant {
   Vote = 'vote',
   Category = 'category',
+  Slim = 'slim',
 }
 
 interface Props {
-  proposal: ProposalAttributes
+  proposal: ProposalAttributes | PriorityProposal
   votes?: Record<string, Vote>
   variant: `${Variant}`
 }
 
+function getPriorityText(proposal: PriorityProposal, t: (...args: any[]) => any) {
+  const isProposalActive = Time().isBefore(Time(proposal.finish_at))
+  const dateText = getDateText(t, isProposalActive, proposal.finish_at)
+
+  switch (proposal.priority_type) {
+    case PriorityProposalType.ActiveGovernance:
+      return dateText
+    case PriorityProposalType.OpenPitch:
+      return 'Open for submissions'
+    case PriorityProposalType.PitchWithSubmissions:
+      return `Tender submissions end in lalala`
+    case PriorityProposalType.PitchOnVotingPhase:
+      return `Tenders open for voting`
+    case PriorityProposalType.OpenTender:
+      return dateText
+    case PriorityProposalType.TenderWithSubmissions:
+      return `Bid submissions end in lalala`
+    case PriorityProposalType.ActiveBid:
+      return dateText
+  }
+}
+
+function getDateText(t: (...args: any[]) => any, isProposalActive: boolean, finish_at: Date) {
+  return t(`page.home.open_proposals.${isProposalActive ? 'ends_date' : 'ended_date'}`, {
```

Comment:
We have a `useProposalDateText` hook that we use in some components. Could we reuse that hook here?